File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 10/3/2025, 2:14:02 PM
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer,Integer> hm=new HashMap<>();

        for(int a: nums){
             hm.put(a,hm.getOrDefault(a,0)+1);
             if(hm.get(a)>1) return true;
        }
        return false;
    }
}

File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 10/3/2025, 2:13:55 PM
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        d={}
        for i in range(len(nums)):
            if nums[i] in d:
                return True
            d[nums[i]]=1
        return False